#Python imports

In [8]:
import numpy as np
import pandas as pd

#datasource

In [9]:
# Load the data from the CSV file into a numpy array
data = pd.read_csv('raw_house_data.csv', header=None)

In [10]:
# Print the shape of the data array to verify that it loaded correctly
print("Data shape:", data.shape)

Data shape: (5001, 16)


The Data_Shape: (5001, 16) output says that the data has a shape of 5001 rows and 16 columns.

What this means in we have 5001 data points and for 1 data point we have 16 features.

#The steps to preprocessing the data

Now the data is loaded in and we understand it's shape, it is time to understand preprocessing. Preprocessing a dataset refers to techniques and methods. The steps are:
- Cleaning the data
- handling missing values
- transforming data for increased suitability for modelling
- and either scaling or normalizing the data.

#Cleaning the data

we use the dropna() method to remove any rows with null values. The resulting DataFrame with null values removed is stored in data_clean.

In [12]:
# Remove rows with any null values
data_clean = data.dropna()

We use the drop_duplicates() method to get rid of any duplicate rows from the dataframe

In [13]:
# Drop duplicates
data_clean = data.drop_duplicates()